File: Front/Timer/src/app/model/Sesion.ts
import { Categoria } from './Categoria';
import { Intento } from './Intento';
import { Usuario } from './Usuario';

export class Sesion {
    id: number;
    nombre: string;
    categoria: Categoria;
    usuario: Usuario;
    intentos: Intento[];

    constructor(
        id: number = 0,
        nombre: string = '',
        categoria: Categoria = new Categoria(),
        usuario: Usuario = new Usuario(),
        intentos: Intento[] = []
    ) {
        this.id = id;
        this.nombre = nombre;
        this.categoria = categoria;
        this.usuario = usuario;
        this.intentos = intentos;
    }
}

File: apiTimer/src/main/java/com/apiTimer/entidades/Sesion.java
package com.apiTimer.entidades;

import jakarta.persistence.*;
import java.io.Serializable;
import java.util.HashSet;
import java.util.Set;

@Entity
@Table(name = "Sesiones")
public class Sesion implements Serializable {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @Column(nullable = false, length = 100)
    private String nombre;

    @ManyToOne
    @JoinColumn(name = "usuario_id", nullable = false)
    private Usuario usuario;

    @OneToMany(mappedBy = "sesion", cascade = CascadeType.ALL, orphanRemoval = true)
    private Set<Intento> intentos = new HashSet<>();

    // Getters and Setters
    public Long getId() {
        return id;
    }

    public void setId(Long id) {
        this.id = id;
    }

    public String getNombre() {
        return nombre;
    }

    public void setNombre(String nombre) {
        this.nombre = nombre;
    }

    public Usuario getUsuario() {
        return usuario;
    }

    public void setUsuario(Usuario usuario) {
        this.usuario = usuario;
    }

    public Set<Intento> getIntentos() {
        return intentos;
    }

    public void setIntentos(Set<Intento> intentos) {
        this.intentos = intentos;
    }
}
